### ISSUE TITLE
ng update 8.0 to 9.1 fails during Undecorated classes migration

### ISSUE THREAD

**Issue by kernwig** [angular/angular#36582]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

No, since updates are version specific.

### Description

Running `ng update @angular/cli @angular/core` on a fairly simple Angular 8.0.3 application fails with a nondescript error during Undecorated classes with decorated fields migration. This one 👉 https://angular.io/guide/migration-undecorated-classes

The only possible related cause I could fine is an abstract base class for page components. I tried decorating it as a `@Component` and making it not abstract, but error remained.

## 🔬 Minimal Reproduction

Without a useful error message, I cannot reduce the problem down to minimal. Fortunately the application is open source and straight forward.

Repository is at https://github.com/kernwig/sqac-amplify
Clone via HTTPS or SSH:
```
git clone https://github.com/kernwig/sqac-amplify.git
git clone git@github.com:kernwig/sqac-amplify.git
```

The `develop` branch is the working Angular 8 app.

The abstract base class component is https://github.com/kernwig/sqac-amplify/blob/develop/src/app/shared/abstract-base.component.ts

## 🔥 Exception or Error

```
dev/sqac/sqac-amplify % ng update @angular/cli @angular/core                                                      
The installed Angular CLI version is older than the latest stable version.
Installing a temporary version to perform the update.
Installing packages for tooling via npm.
Installed packages for tooling via npm.
Using package manager: 'npm'
Collecting installed dependencies...
Found 39 dependencies.
Fetching dependency metadata from registry...
                  Package "@angular/core" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/router" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/animations" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/forms" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/platform-browser-dynamic" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/platform-browser" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/compiler-cli" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/compiler" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/common" has a missing peer dependency of "tslib" @ "^1.10.0".
                  Package "@angular/service-worker" has a missing peer dependency of "tslib" @ "^1.10.0".
    Updating package.json with dependency @angular/cli @ "9.1.1" (was "8.3.21")...
    Updating package.json with dependency @angular/core @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular-devkit/build-angular @ "0.901.1" (was "0.803.21")...
    Updating package.json with dependency @angular/router @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/animations @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/forms @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/platform-browser-dynamic @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/platform-browser @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/compiler-cli @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/compiler @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/common @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency @angular/service-worker @ "9.1.1" (was "8.2.14")...
    Updating package.json with dependency zone.js @ "0.10.3" (was "0.9.1")...
    Updating package.json with dependency typescript @ "3.8.3" (was "3.4.5")...
UPDATE package.json (2140 bytes)
✔ Packages installed successfully.
** Executing migrations of package '@angular/cli' **

❯ Angular Workspace migration.
  Update an Angular CLI workspace to version 9.
UPDATE angular.json (4486 bytes)
UPDATE src/ngsw-config.json (830 bytes)
UPDATE src/tsconfig.app.json (229 bytes)
UPDATE package.json (2164 bytes)
✔ Packages installed successfully.
  Migration completed.

❯ Lazy loading syntax migration.
  Update lazy loading syntax to use dynamic imports.
  Migration completed.

❯ Replace deprecated 'styleext' and 'spec' Angular schematic options.
UPDATE angular.json (4511 bytes)
  Migration completed.

** Executing migrations of package '@angular/core' **

❯ Static flag migration.
  Removes the `static` flag from dynamic queries.
  As of Angular 9, the "static" flag defaults to false and is no longer required for your view and content queries.
  Read more about this here: https://v9.angular.io/guide/migration-dynamic-flag
UPDATE src/app/collections/widget/searchable-input.component.ts (5771 bytes)
  Migration completed.

❯ Missing @Injectable and incomplete provider definition migration.
  In Angular 9, enforcement of @Injectable decorators for DI is a bit stricter and incomplete provider definitions behave differently.
  Read more about this here: https://v9.angular.io/guide/migration-injectable
  Migration completed.

❯ ModuleWithProviders migration.
  In Angular 9, the ModuleWithProviders type without a generic has been deprecated.
  This migration adds the generic where it is missing.
  Read more about this here: https://v9.angular.io/guide/migration-module-with-providers
  Migration completed.

❯ Renderer to Renderer2 migration.
  As of Angular 9, the Renderer class is no longer available.
  Renderer2 should be used instead.
  Read more about this here: https://v9.angular.io/guide/migration-renderer
UPDATE src/app/widgets/bootstrap-switch.component.ts (14462 bytes)
  Migration completed.

❯ Undecorated classes with decorated fields migration.
  As of Angular 9, it is no longer supported to have Angular field decorators on a class that does not have an Angular decorator.
  Read more about this here: https://v9.angular.io/guide/migration-undecorated-classes
✖ Migration failed: Cannot read property 'kind' of undefined
  See "/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/ng-xPo1tT/angular-errors.log" for further details.

dev/sqac/sqac-amplify % cat /private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/ng-xPo1tT/angular-errors.log
[error] TypeError: Cannot read property 'kind' of undefined
    at Object.isClassDeclaration (/Users/adamfanello/dev/sqac/sqac-amplify/node_modules/typescript/lib/typescript.js:11439:21)
    at Object.findBaseClassDeclarations (/Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/utils/typescript/find_base_classes.js:31:32)
    at /Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/migrations/undecorated-classes-with-decorated-fields/transform.js:83:71
    at Set.forEach (<anonymous>)
    at UndecoratedClassesWithDecoratedFieldsTransform._findUndecoratedAbstractDirectives (/Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/migrations/undecorated-classes-with-decorated-fields/transform.js:82:32)
    at UndecoratedClassesWithDecoratedFieldsTransform.migrate (/Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/migrations/undecorated-classes-with-decorated-fields/transform.js:41:18)
    at runUndecoratedClassesMigration (/Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/migrations/undecorated-classes-with-decorated-fields/index.js:49:19)
    at /Users/adamfanello/dev/sqac/sqac-amplify/node_modules/@angular/core/schematics/migrations/undecorated-classes-with-decorated-fields/index.js:37:17
    at MergeMapSubscriber.project (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/@angular-devkit/schematics/src/rules/call.js:74:24)
    at MergeMapSubscriber._tryNext (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/operators/mergeMap.js:69:27)
    at MergeMapSubscriber._next (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/operators/mergeMap.js:59:18)
    at MergeMapSubscriber.Subscriber.next (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/Subscriber.js:66:18)
    at Observable._subscribe (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/util/subscribeToArray.js:5:20)
    at Observable._trySubscribe (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/Observable.js:44:25)
    at Observable.subscribe (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/Observable.js:30:22)
    at MergeMapOperator.call (/private/var/folders/vp/4nr1pk5n0mxf2qyy342k_g7h0000gn/T/.ng-temp-packages-8LHoul/node_modules/rxjs/internal/operators/mergeMap.js:39:23)
```

## 🌍 Your Environment

This is the pre-attempt version:
<pre><code>
Angular CLI: 8.3.25
Node: 12.16.1
OS: darwin x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.803.25
@angular-devkit/build-optimizer   0.803.25
@angular-devkit/build-webpack     0.803.25
@angular-devkit/core              8.3.25
@angular-devkit/schematics        8.3.25
@angular/cli                      8.3.25
@ngtools/webpack                  8.3.25
@schematics/angular               8.3.25
@schematics/update                0.803.25
rxjs                              6.5.4
typescript                        3.4.5
webpack                           4.39.2
</code></pre>

Post-attempt:
```
Angular CLI: 9.1.1
Node: 12.16.1
OS: darwin x64

Angular: 9.1.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
... service-worker
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.1
@angular-devkit/build-angular     0.901.1
@angular-devkit/build-optimizer   0.901.1
@angular-devkit/build-webpack     0.901.1
@angular-devkit/core              9.1.1
@angular-devkit/schematics        9.1.1
@ngtools/webpack                  9.1.1
@schematics/angular               9.1.1
@schematics/update                0.901.1
rxjs                              6.5.4
typescript                        3.8.3
webpack                           4.42.0
```

**Anything else relevant?**

This is the same migration attempt that I reported issue #36346. Thank you for fixing that one - but now it fails here.

**Comment by JoostK** [comment]
Duplicate of #36522 which has been fixed in #36543, will be available in the next release.

**Comment by devversion** [comment]
Yep. Just wanted to get back to this too. Fix landed with https://github.com/angular/angular/commit/ca677481a2ac3675b84698e2d16fd379e1799e2c. Thanks for reporting back!
